src/server/routers/lambda/user.ts

```diff
@@ -14,7 +16,38 @@ const userProcedure = authedProcedure.use(async (opts) => {
 
 export const userRouter = router({
   getUserState: userProcedure.query(async ({ ctx }): Promise<UserInitializationState> => {
-    const state = await ctx.userModel.getUserState(ctx.userId);
+    let state: Awaited<ReturnType<UserModel['getUserState']>> | undefined;
+
+    // get or create first-time user
+    while (!state) {
+      try {
+        state = await ctx.userModel.getUserState(ctx.userId);
+      } catch (error) {
+        if (enableClerk && error instanceof UserNotFoundError) {
+          const user = await currentUser();
+          if (user) {
+            const email =
+              user.primaryEmailAddress ??
```

Comment:
I think the clerkUser pick logical can be the same with the user creating webooks? 

https://github.com/lobehub/lobe-chat/blob/a1e8371bbe77a0d62876d1df70af24cfb7d44cc1/src/app/api/webhooks/clerk/route.ts#L30-L42

maybe we can add a method named `createUserByClerkUser(clerkUser:ClerkUser,otherParams)` in user model. Then we can just pass the clerkUser as a params

Reply:
I tried but they're actually 2 different types (`User` vs. `UserJSON`) and are named differently (camelCase vs. snake_case). If we do this, we'll also need to convert Clerk `User` to `UserJSON` which is not better than converting it into `NewUser` directly in my opinion.

`@clerkjs/backend` has [built-in function to create an `User` instance from a `UserJSON`](https://github.com/clerk/javascript/blob/main/packages/backend/src/api/resources/User.ts) but unfortunately it's not exported yet, I created a feedback for them to export it, then we can use this function and merge the logic.

Reply:
It will probably take a while for the Clerk team to react to the feedback, so ideally we merge this and update later when Clerk updates their SDKs.